### PR TITLE
[MRG] update the sourmash __init__.py docstring

### LINKED ISSUE #1233: improve sourmash top-level docstring
it would be nice to better support e.g. Jupyter Notebook introspection of sourmash top-level commands

### PR DESCRIPTION
**Pull request by ctb**
Adds a better docstring and `__all__` to `src/sourmash/__init__.py`.

Fixes https://github.com/dib-lab/sourmash/issues/1233.

## Checklist

- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review @luizirber @bluegenes

File: src/sourmash/__init__.py [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
+The sourmash code is available at github.com/dib-lab/sourmash/ under the
+BSD 3-Clause license.
 """
-import re
-import math
-import os
 from deprecation import deprecated
 
+__all__ = ['MinHash', 'SourmashSignature',
+           'load_one_signature',
```
Comment (bluegenes) [inline src/sourmash/__init__.py:21]: is `load_one_signature` also deprecated / to be removed in 5.0?

**Review by bluegenes — APPROVED** [review]
I don't think I can properly evaluate additional things that might be useful here .. but this seems more informative/useful than before, makes sense, looks good

File: src/sourmash/__init__.py (hunk 1 above)
Comment (ctb) [reply src/sourmash/__init__.py:21]: no - it's used too widely/is too useful - see https://github.com/dib-lab/sourmash/issues/1062 for discussion.

I know it's a little inconsistent but I went for practicality here ;).

ref https://github.com/dib-lab/sourmash/pull/1279

**Comment by ctb** [comment]
thanks @bluegenes! @luizirber lmk if you have additional thoughts, no particular hurry on merging this.

**Pull request merged**